### DISCUSSION TITLE
How to Install TIGRE 2.6 In Window.

### DISCUSSION THREAD

**Discussion by kbhyun12** [CERN/TIGRE#567] [Q&A]
I wanted to apply the TIGRE 2.6 library in a Windows environment using Python 3.9 and PyCharm IDE, CUDA 12.2. However, it says that the _AX module is missing. Please let me know how to install it

**Comment by AnderBiguri** [comment]
Hi. Unfortunately with that information I cna only say that you have not properly installed it, but I can't k ow why.

Have you followed the installation instructions?

**Comment by kbhyun12** [comment]
Hi.

First, thank you for your response. However, I am still experiencing difficulties with the installation. I followed the steps for Windows installation, but I am encountering the following error.



-----Original Message-----

**Reply by AnderBiguri** [reply]
There is no error in your message :)

**Reply by kbhyun12** [reply]
![Error](https://github.com/user-attachments/assets/869ef8d2-1f40-49bd-9434-7d1ecf76a824)
Please Check Again.

**Reply by AnderBiguri** [reply]
The error says that you are using a unsupported Microsoft visual studio code. It does suggest which range to install. Have you tried installing one of the ones suggested by the error?

**Reply by kbhyun12** [reply]
Yes! I have already installed MSVC 2015 and 2022.

**Reply by AnderBiguri** [reply]
The message suggests that indeed 2015 is not valid, as it says 2017 to 2022 (inclusive). My guess is that at installing 2022 you have not uninstalled the other one, or you have not installed the C++ compiler, or that the enviroment variables still point to the old one.

**Comment by TheHorizonSeeker** [comment]
I got a similar problem when installing tigre python version, and my versions are python 3.11, cuda 11.8, MS 2022. The console showed I installed pytigre successfully, but when I run d03.py, it shows: ImportError: DLL load failed while importing _Ax.  I haven't solved this problem yet!

**Reply by AnderBiguri** [reply]
Try the following: https://github.com/CERN/TIGRE/issues/628#issuecomment-2595064024

**Reply by TheHorizonSeeker** [reply]
Thanks a lot, I solve this problem and it works fine.

**Reply by TheHorizonSeeker** [reply]
One more question,  for CBCT, the object is fixed and the source and detector is rotating around the object in astra tool box, but in TIGRE, it seems like the source and detector are fixed, the object rotates on the Z axis, is it right? This is critical because I wanna design a new CT sampling method. Thanks!

**Reply by AnderBiguri** [reply]
@TheHorizonSeeker Its not critical at all ;) the difference between having a rotating stage or a rotating machine is `angles=-angles`. The position of everything is relative to each other. I have reconstructed with TIGRE data from both cases.

**Reply by TheHorizonSeeker** [reply]
I understand what you mean, but when I write a new matlab script in which the detector self-rotates:
```
ax = linspace(0, 180, n)*pi/180;
ay = linspace(ang, ang, n)*pi/180;
az = linspace(0, 0, n)*pi/180;
geo.rotDetector=[ax; ay; az];
```
it likes I just tilt the rotation axis of the detector rather than move the detector to other positions because the rotDetector is  "Rotation of the detector, by X,Y and Z axis respectively."
And I feel a little confused about angles and rotDetector, I try to see the projections plot to validate my thoughts.

**Reply by AnderBiguri** [reply]
@TheHorizonSeeker that rotation is irrelevant to the machine/object moving. Its the rotation, _in place_ of the detector, such that it becomes non-perpendicular to the source-detector line. Those angles are not rotation angles of the machine, but the direction of the detector normal vector, in other words. A normal circular scanner will alwasy have these be zero, for every gantry rotation angle. 

You can instead give the angles variable 3 inputs, that are ZYZ Euler angles, to rotate the gantry in a spherical fashion around the object.

**Reply by TheHorizonSeeker** [reply]
@AnderBiguri Thanks for clearing these up, you guys really provide a valuable promotion for CT science^_^
